packages/upload-media/README.md

```diff
@@ -0,0 +1,147 @@
+-   _$0.onError_ `[AddItemsArgs[ 'onError' ]]`: Function called when an error happens.
+-   _$0.additionalData_ `[AddItemsArgs[ 'additionalData' ]]`: Additional data to include in the request.
+
+#### cancelItem
+
+Cancels an item in the queue based on an error.
+
+_Parameters_
+
+-   _id_ `QueueItemId`: Item ID.
+-   _error_ `Error`: Error instance.
+-   _silent_ Whether to cancel the item silently, without invoking its `onError` callback.
+
+<!-- END TOKEN(Autogenerated actions|src/store/actions.ts) -->
+
+### Selectors
+
+The following selectors are available on the object returned by `wp.data.select( 'core/upload-media' )`:
+
+<!-- START TOKEN(Autogenerated selectors|src/store/selectors.ts) -->
+
+#### getComparisonDataForApproval
+
+Returns data to compare the old file vs. the optimized file, given the attachment ID.
+
+Includes both the URLs and the respective file sizes and the size difference in percentage.
+
+_Parameters_
+
+-   _state_ `State`: Upload state.
+-   _attachmentId_ `number`: Attachment ID.
+
+_Returns_
+
+-   `{ oldUrl: string | undefined; oldSize: number; newSize: number; newUrl: string | undefined; sizeDiff: number; } | null`: Comparison data.
+
+#### getItems
+
+Returns all items currently being uploaded, without sub-sizes (children).
+
+_Parameters_
+
+-   _state_ `State`: Upload state.
+
+_Returns_
+
+-   `QueueItem[]`: Queue items.
+
+#### getSettings
+
+Returns the media upload settings.
+
+_Parameters_
+
+-   _state_ `State`: Upload state.
+
+_Returns_
+
+-   `Settings`: Settings
+
+#### isPendingApproval
+
+Determines whether there is an item pending approval.
+
+_Parameters_
+
+-   _state_ `State`: Upload state.
+
+_Returns_
+
+-   `boolean`: Whether there is an item pending approval.
+
+#### isPendingApprovalByAttachmentId
+
+Determines whether an item is the first one pending approval given its associated attachment ID.
+
+_Parameters_
+
+-   _state_ `State`: Upload state.
+-   _attachmentId_ `number`: Attachment ID.
+
+_Returns_
+
+-   `boolean`: Whether the item is first in the list of items pending approval.
+
+#### isUploading
+
+Determines whether any upload is currently in progress.
+
+_Parameters_
+
+-   _state_ `State`: Upload state.
+
+_Returns_
+
+-   `boolean`: Whether any upload is currently in progress.
+
+#### isUploadingById
+
+Determines whether an upload is currently in progress given an attachment ID.
+
+_Parameters_
+
+-   _state_ `State`: Upload state.
+-   _attachmentId_ `number`: Attachment ID.
+
+_Returns_
+
+-   `boolean`: Whether upload is currently in progress for the given attachment.
+
+#### isUploadingByUrl
+
+Determines whether an upload is currently in progress given an attachment URL.
+
+_Parameters_
+
+-   _state_ `State`: Upload state.
+-   _url_ `string`: Attachment URL.
+
+_Returns_
+
+-   `boolean`: Whether upload is currently in progress for the given attachment.
+
+<!-- END TOKEN(Autogenerated selectors|src/store/selectors.ts) -->
```

Comment:
Are all these public selectors something we expect users to be able to use or are these internal to the package.